assets/app/scripts/directives/logViewer.js

```diff
@@ -64,6 +64,52 @@ angular.module('openshiftConsole')
               logLinks.scrollTop();
             };
 
+            var bufferedLines = [];
```

Comment:
What if you used jquery to build a template for the log line once (more readable, but worse for performance), then clone it using cloneNode(true) inside the loop, and set the line number text and log text inside the cloned line and append it to the fragment